src/controllers/authController.ts

```diff
@@ -21,55 +22,26 @@ export const createUser = async (req: Request, res: Response) => {
         verificationCode: code,
     });
 
-    try {
-        const emailCheck = await User.findOne({
-            $or: [{ email: newUser.email }],
-        });
-        if (emailCheck) {
-            return res.status(400).send({
-                message:
-                    "This Email is already in use, please login to your account",
-            });
-        }
+    const check = await findUser("email", newUser.email);
+    if (check) return res.status(400).json("User already exists, please login");
```

Comment:
How are u sure the user that's trying to register is the same user that has his/her details in your database?...Messaghe like "User already exist" is fine. Please remove the rest